### PR TITLE
SystemError: Fix inconsistent signature.

### PR DESCRIPTION
**Pull request by yxhuvud**
In `.from_os_error` the code looks like
```
message = self.build_message(message, **opts)
message =
  if message
    ..
  else
    ..
  end
```
which clearly doesn't make sense if `build_message` always return a
string.

### CODE REVIEW

File: spec/std/system_error_spec.cr [hunk 1]
```diff
@@ -0,0 +1,11 @@
+require "spec"
+
+describe SystemError do
+  describe ".from_os_error" do
+    it "Can build an error from an errno" do
+      errno = Errno.new(11)
+      error = ::IO::Error.from_os_error(message: nil, os_error: errno)
+      error.message.should eq("Resource temporarily unavailable")
```
Comment (yxhuvud) [inline spec/std/system_error_spec.cr:8]: LibC-specific. It will be interesting to see if it passes CI.

File: src/system_error.cr [hunk 2]
```diff
@@ -85,7 +85,7 @@ module SystemError
     #
     # By default it returns the original message unchanged. But that could be
     # customized based on the keyword arguments passed to `from_errno` or `from_winerror`.
-    protected def build_message(message, **opts) : String
+    protected def build_message(message, **opts) : String?
```
Comment (straight-shoota) [inline src/system_error.cr:88]: Could also add `String?` restriction to `message` like in `from_os_error`.

**Review by straight-shoota — APPROVED** [review]

File: src/system_error.cr (hunk 2 above)
Comment (yxhuvud) [reply src/system_error.cr]: Right. That won't affect functionality but I guess it ends up clearer.

File: spec/std/system_error_spec.cr (hunk 1 above)
Comment (yxhuvud) [reply spec/std/system_error_spec.cr:8]: It doesn't. On Mac it gives `"Resource deadlock avoided"`. I'll change to match against /Resource/ unless someone has a better idea.

Comment (yxhuvud) [reply spec/std/system_error_spec.cr]: BTW, it is a bit strange to get email about failed specs, but that it doesn't show up as broken in the github UI. Is that a known issue?

Comment (straight-shoota) [reply spec/std/system_error_spec.cr]: Maybe the CI notification is for an older commit?

Comment (yxhuvud) [reply spec/std/system_error_spec.cr]: Can't be, I got two failure notifications and no obvious sign in the UI. But perhaps it was there but just not obvious enough :shrug:

Comment (straight-shoota) [reply spec/std/system_error_spec.cr]: For the error messages, you can just compare the respective manpages (https://man7.org/linux/man-pages/man3/errno.3.html, https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man2/intro.2.html, https://www.freebsd.org/cgi/man.cgi?query=errno&sektion=2&format=html, etc.).

I suppose `ENOENT` should be pretty portable: `No such file or directory`.

**Review by beta-ziliani — APPROVED** [review]
Thanks @yxhuvud !

**Pull request merged**